### PR TITLE
REST service

### PR DESCRIPTION
**Pull request by SignatureBeef**
Adds an Owin REST service
Configurations now save defaults when created
Adds the Service attribute to PlanTextAccountService

### CODE REVIEW

File: Orion/Rest/Owin/OwinRestService.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+			Debug.Assert(_webApp != null);
+			_webApp.Dispose();
+			_webApp = null;
+		}
+
+		/// <summary>
+		/// Occurs when OWIN requests the service to configure the HTTP service.
+		/// </summary>
+		/// <param name="appBuilder">The <see cref="IAppBuilder"/> instance for use with configuring the HTTP service</param>
+		private void Startup(IAppBuilder appBuilder)
```
Comment (Ijwu) [inline Orion/Rest/Owin/OwinRestService.cs:83]: Isn't the use of IAppBuilder deprecated? I can't find any meaningful documentation about the interface itself.

Comment (SignatureBeef) [inline Orion/Rest/Owin/OwinRestService.cs:83]: @Ijwu You might be right as the docs I was looking at were back from 2013.
It seems ASP .NET Core is where they are heading to (?) which uses IApplicationBuilder. 
If we're on the same page and we don't mind using the mvc packages for the web api + a billion references, I can easily switch this over tonight.
See https://docs.asp.net/en/latest/intro.html

**Comment by QuiCM** [comment]
Tagging in @NyxStudios/tshock for further commentary on this. Are we all good for a conflict resolution + merge?

**Comment by QuiCM** [comment]
@DeathCradle does this now fulfil all of the requests mentioned above?

**Comment by SignatureBeef** [comment]
@WhiteXZ The stuff mentioned from @tylerjwatson, yeah it is.
The .NET Core stuff ended up causing too many problems so after a discussion with @tylerjwatson we decided that reverting back to the owin service was best.

**Pull request merged**